tests/ui/coroutine/clone-impl.rs

```diff
@@ -42,6 +42,7 @@ fn test3_upvars() {
     let clonable_0: Vec<u32> = Vec::new();
     let gen_clone_0 = #[coroutine]
     move || {
+        yield;
```

Comment:
why this?

Reply:
Because otherwise it is ambiguous because we never constrain the yield type. 

Since we now stall based on coroutine and not witness, we end up delaying the error for when an upvar is not cloneable until after we would've errored on unconstrained ty vars.

For contrast, observe that other coroutines in this file also constrain their yield type by using a dummy yield statement.